pkg/config/remote/service/service.go

```diff
@@ -173,37 +209,93 @@ func WithDatabaseFileName(fileName string) func(s *options) {
 	options := defaultOptions
 	for _, opt := range opts {
 		opt(&options)
 	}
 
-	refreshIntervalOverrideAllowed := false // If a user provides a value we don't want to override
-	var refreshInterval time.Duration
-	if cfg.IsSet("remote_configuration.refresh_interval") {
-		refreshInterval = cfg.GetDuration("remote_configuration.refresh_interval")
-	} else {
-		refreshIntervalOverrideAllowed = true
-		refreshInterval = defaultRefreshInterval
+	refreshInterval := defaultRefreshInterval
+	refreshIntervalOverrideAllowed := true
+	if options.refresh != nil && options.refresh.interval >= minimalRefreshInterval {
+		refreshInterval = options.refresh.interval
+		refreshIntervalOverrideAllowed = false
+	} else if options.refresh != nil {
+		log.Warnf("[%s] %s is set to %v which is below the minimum of %v - using default refresh interval %v", options.flavor, options.refresh.cfgPath, options.refresh.interval, minimalRefreshInterval, defaultRefreshInterval)
 	}
 
-	// Either invalid (which resolves to 0) or was explicitly set below minimal. If it was invalid there would
-	// be an additional error message describing the failure to parse the value.
-	if refreshInterval < minimalRefreshInterval {
-		log.Warnf("remote_configuration.refresh_interval is set to %v which is below the minimum of %v - using default refresh interval %v", refreshInterval, minimalRefreshInterval, defaultRefreshInterval)
-		refreshInterval = defaultRefreshInterval
-		refreshIntervalOverrideAllowed = true
-	}
-
-	maxBackoffTime := cfg.GetDuration("remote_configuration.max_backoff_interval")
-	if maxBackoffTime < minimalMaxBackoffTime {
-		log.Warnf("remote_configuration.max_backoff_time is set to %v which is below the minimum of %v - setting value to %v", maxBackoffTime, minimalMaxBackoffTime, minimalMaxBackoffTime)
-		maxBackoffTime = minimalMaxBackoffTime
-	} else if maxBackoffTime > maximalMaxBackoffTime {
-		log.Warnf("remote_configuration.max_backoff_time is set to %v which is above the maximum of %v - setting value to %v", maxBackoffTime, maximalMaxBackoffTime, maximalMaxBackoffTime)
-		maxBackoffTime = maximalMaxBackoffTime
+	maxBackoffTime := minimalMaxBackoffTime
+	if options.maxBackoff != nil {
+		if options.maxBackoff.interval < minimalMaxBackoffTime {
+			log.Warnf("[%s] %s is set to %v which is below the minimum of %v - setting value to %v", options.flavor, options.maxBackoff.cfgPath, options.maxBackoff.interval, minimalMaxBackoffTime, minimalMaxBackoffTime)
+			maxBackoffTime = minimalMaxBackoffTime
+		} else if options.maxBackoff.interval > maximalMaxBackoffTime {
+			log.Warnf("[%s] %s is set to %v which is above the maximum of %v - setting value to %v", options.flavor, options.maxBackoff.cfgPath, options.maxBackoff.interval, maximalMaxBackoffTime, maximalMaxBackoffTime)
+			maxBackoffTime = maximalMaxBackoffTime
+		} else {
+			maxBackoffTime = options.maxBackoff.interval
+		}
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ Question:** ‏Shouldn't we do verifications like these in the methods returned by the options, directly?